File: examples/ios/ExploreScreen.swift
import SwiftUI
import PurpleUI

/// Explore tab — search bar + bottom tray + map pins over a simulated map background.
///
/// This pattern matches the Purple app's main map flow:
/// - `PUSearchBar` at the top for location / query input
/// - `PUFloatingButton` for centering on current location
/// - `PUBottomTray` displaying nearby WiFi spots
/// - `PUMapPin` markers on the map canvas
struct ExploreScreen: View {
    @State private var searchQuery = ""
    @State private var selectedPin: Int? = nil
    @State private var showToast = false

    private let pins: [(x: CGFloat, y: CGFloat)] = [
        (0.25, 0.35), (0.55, 0.25), (0.70, 0.50), (0.40, 0.60)
    ]

    var body: some View {
        ZStack(alignment: .bottom) {
            // ── Simulated map background ──────────────────────────────────
            mapCanvas

            // ── Search bar ────────────────────────────────────────────────
            VStack {
                PUSearchBar(text: $searchQuery, placeholder: "Search WiFi spots…")
                    .padding(.horizontal, PUSpacing.lg)
                    .padding(.top, 60)
                Spacer()
            }

            // ── Floating nav button ───────────────────────────────────────
            VStack {
                Spacer()
                HStack {
                    Spacer()
                    PUFloatingButton(variant: .icon, icon: .navigate, dark: true) {
                        showToast = true
                    }
                    .padding(.trailing, PUSpacing.lg)
                    .padding(.bottom, 260)
                }
            }

            // ── Bottom tray ───────────────────────────────────────────────
            PUBottomTray(title: "34 WiFi nearby", dark: true, peekHeight: 200, expandHeight: 480) {
                ScrollView {
                    VStack(spacing: 0) {
                        ForEach(0..<8) { i in
                            trayRow(index: i)
                        }
                    }
                }
            }

            // ── Toast ─────────────────────────────────────────────────────
            if showToast {
                VStack {
                    PUToast(message: "Centred on your location", style: .info)
                        .padding(.horizontal, PUSpacing.lg)
                        .padding(.top, 120)
                    Spacer()
                }
                .transition(.move(edge: .top).combined(with: .opacity))
                .onAppear {
                    Task {
                        try? await Task.sleep(nanoseconds: 2_500_000_000)
                        withAnimation { showToast = false }
                    }
                }
            }
        }
        .ignoresSafeArea(edges: .bottom)
        .animation(.easeInOut(duration: 0.25), value: showToast)
    }

    // MARK: - Map canvas with pins

    private var mapCanvas: some View {
        GeometryReader { geo in
            ZStack {
                Color(hex: "#1a2744").ignoresSafeArea()

                // Decorative grid lines
                Path { p in
                    for i in 0..<10 {
                        let x = geo.size.width * CGFloat(i) / 10
                        p.move(to: CGPoint(x: x, y: 0))
                        p.addLine(to: CGPoint(x: x, y: geo.size.height))
                    }
                    for i in 0..<14 {
                        let y = geo.size.height * CGFloat(i) / 14
                        p.move(to: CGPoint(x: 0, y: y))
                        p.addLine(to: CGPoint(x: geo.size.width, y: y))
                    }
                }
                .stroke(Color.white.opacity(0.05), lineWidth: 1)

                // Map pins
                ForEach(pins.indices, id: \.self) { i in
                    PUMapPin(
                        variant: i == 0 ? .teardrop : .dot,
                        selected: selectedPin == i,
                        dark: true
                    )
                    .position(
                        x: geo.size.width  * pins[i].x,
                        y: geo.size.height * pins[i].y
                    )
                    .onTapGesture {
                        withAnimation { selectedPin = selectedPin == i ? nil : i }
                    }
                }
            }
        }
        .ignoresSafeArea()
    }

    // MARK: - Tray row

    private func trayRow(index: Int) -> some View {
        HStack(spacing: PUSpacing.md) {
            Circle()
                .fill(PUColors.brand.opacity(0.2))
                .frame(width: 40, height: 40)
                .overlay(
                    FAIcon(.wifi, size: 16, color: PUColors.brand)
                )

            VStack(alignment: .leading, spacing: 2) {
                Text("WiFi Spot \(index + 1)")
                    .font(PUTypography.bodyBold)
                    .foregroundColor(.white)
                Text("\(Int.random(in: 10...500))m away")
                    .font(PUTypography.supporting)
                    .foregroundColor(.white.opacity(0.5))
            }

            Spacer()

            Text("Free")
                .font(PUTypography.supporting)
                .foregroundColor(PUColors.brand)
        }
        .padding(.horizontal, PUSpacing.xl)
        .padding(.vertical, PUSpacing.md)
    }
}

File: examples/android/ExploreScreen.kt
package com.example.purpleuiexample

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.material3.Text
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp
import ai.purple.purpleui.components.*
import ai.purple.purpleui.icons.FA
import ai.purple.purpleui.icons.FAIcon
import ai.purple.purpleui.tokens.*

/**
 * Explore tab — search bar + bottom tray + map pins over a simulated map background.
 *
 * Demonstrates the main Purple app map flow:
 * - [PUSearchBar] at top for location / query input
 * - [PUFloatingButton] for centering on current location
 * - [PUBottomTray] listing nearby WiFi spots
 * - [PUToastHost] for ephemeral feedback
 */
@Composable
fun ExploreScreen(modifier: Modifier = Modifier) {
    var searchQuery by remember { mutableStateOf("") }
    var selectedPin by remember { mutableStateOf<Int?>(null) }
    var showToast by remember { mutableStateOf(false) }

    PUToastHost(
        isVisible = showToast,
        message = "Centred on your location",
        style = PUToastStyle.Info,
        onDismiss = { showToast = false },
        modifier = modifier.fillMaxSize()
    ) {
        Box(modifier = Modifier.fillMaxSize()) {

            // ── Simulated map background ──────────────────────────────
            MapCanvas(
                selectedPin = selectedPin,
                onPinTap = { i -> selectedPin = if (selectedPin == i) null else i },
                modifier = Modifier.fillMaxSize()
            )

            // ── Search bar ────────────────────────────────────────────
            PUSearchBar(
                value = searchQuery,
                onValueChange = { searchQuery = it },
                placeholder = "Search WiFi spots…",
                showFilter = true,
                modifier = Modifier
                    .fillMaxWidth()
                    .align(Alignment.TopCenter)
                    .statusBarsPadding()
                    .padding(horizontal = PUSpacing.lg, vertical = PUSpacing.lg)
            )

            // ── Floating locate button ────────────────────────────────
            PUFloatingButton(
                icon = PUFloatingButtonIcon.Navigate,
                dark = true,
                modifier = Modifier
                    .align(Alignment.BottomEnd)
                    .padding(end = PUSpacing.lg, bottom = 280.dp),
                onClick = { showToast = true }
            )

            // ── Bottom tray ───────────────────────────────────────────
            PUBottomTray(
                title = "34 WiFi nearby",
                dark = true,
                peekHeight = 200.dp,
                expandHeight = 480.dp,
                modifier = Modifier.align(Alignment.BottomCenter)
            ) {
                LazyColumn(modifier = Modifier.fillMaxSize()) {
                    items(8) { i -> TrayRow(index = i) }
                }
            }
        }
    }
}

// ---------------------------------------------------------------------------

@Composable
private fun MapCanvas(
    selectedPin: Int?,
    onPinTap: (Int) -> Unit,
    modifier: Modifier = Modifier
) {
    val pins = listOf(
        0.25f to 0.35f,
        0.55f to 0.25f,
        0.70f to 0.50f,
        0.40f to 0.60f
    )

    Box(modifier = modifier.background(Color(0xFF1A2744))) {
        BoxWithConstraints(modifier = Modifier.fillMaxSize()) {
            val w = maxWidth
            val h = maxHeight
            pins.forEachIndexed { i, (xFrac, yFrac) ->
                val variant = if (i == 0) PUFloatingButtonVariant.Icon else PUFloatingButtonVariant.Pill
                Box(
                    modifier = Modifier
                        .offset(x = w * xFrac - 15.dp, y = h * yFrac - 15.dp)
                        .size(30.dp)
                        .clip(CircleShape)
                        .background(
                            if (selectedPin == i) PUColors.brand else PUColors.brand.copy(alpha = 0.6f)
                        )
                        .clickable { onPinTap(i) },
                    contentAlignment = Alignment.Center
                ) {
                    FAIcon(FA.wifi, size = 14.sp, color = Color.White)
                }
            }
        }
    }
}

@Composable
private fun TrayRow(index: Int) {
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .padding(horizontal = PUSpacing.xl, vertical = PUSpacing.md),
        horizontalArrangement = Arrangement.spacedBy(PUSpacing.md),
        verticalAlignment = Alignment.CenterVertically
    ) {
        Box(
            modifier = Modifier
                .size(40.dp)
                .clip(CircleShape)
                .background(PUColors.brand.copy(alpha = 0.2f)),
            contentAlignment = Alignment.Center
        ) {
            FAIcon(FA.wifi, size = 16.sp, color = PUColors.brand)
        }

        Column(verticalArrangement = Arrangement.spacedBy(2.dp), modifier = Modifier.weight(1f)) {
            Text("WiFi Spot ${index + 1}", style = PUTypography.bodyBold, color = Color.White)
            Text("${(10..500).random()}m away", style = PUTypography.supporting, color = Color.White.copy(alpha = 0.5f))
        }

        Text("Free", style = PUTypography.supporting, color = PUColors.brand)
    }
}

private val Int.sp get() = this.toFloat().let { androidx.compose.ui.unit.TextUnit(it, androidx.compose.ui.unit.TextUnitType.Sp) }
private val Float.dp get() = androidx.compose.ui.unit.Dp(this)
